js/edge.js

```diff
@@ -0,0 +1,816 @@
+   */
+  setAdbPort(port) {
+    return this.addArguments('--adb-port=' + port);
+  }
+
+  /**
+   * Sets the path of the log file the driver should log to. If a log file is
+   * not specified, the driver will log to stderr.
+   * @param {string} path Path of the log file to use.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  loggingTo(path) {
+    return this.addArguments('--log-path=' + path);
+  }
+
+  /**
+   * Enables verbose logging.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  enableVerboseLogging() {
+    return this.addArguments('--verbose');
+  }
+
+  /**
+   * Sets the number of threads the driver should use to manage HTTP requests.
+   * By default, the driver will use 4 threads.
+   * @param {number} n The number of threads to use.
+   * @return {!ServiceBuilder} A self reference.
+   */
+  setNumHttpThreads(n) {
+    return this.addArguments('--http-threads=' + n);
+  }
+
+  /**
+   * @override
+   */
+  setPath(path) {
+    super.setPath(path);
+    return this.addArguments('--url-base=' + path);
+  }
+}
+
+
+/** @type {remote.DriverService} */
+var defaultService = null;
+
+
+/**
+ * Sets the default service to use for new MicrosoftEdgeDriver instances.
+ * @param {!remote.DriverService} service The service to use.
+ * @throws {Error} If the default service is currently running.
+ */
+function setDefaultService(service) {
+  if (defaultService && defaultService.isRunning()) {
+    throw Error(
+      'The previously configured EdgeDriver service is still running. ' +
+      'You must shut it down before you may adjust its configuration.');
+  }
+  defaultService = service;
+}
+
+
+/**
+ * Returns the default MicrosoftEdgeDriver service. If such a service has
+ * not been configured, one will be constructed using the default configuration
+ * for an MicrosoftEdgeDriver executable found on the system PATH.
+ * @return {!remote.DriverService} The default MicrosoftEdgeDriver service.
+ */
+function getDefaultService(useEdgeChromium) {
+  if (!defaultService) {
+    defaultService = new ServiceBuilder('', useEdgeChromium).build();
+  }
+  return defaultService;
+}
+
+
+/**
+ * Creates a new WebDriver client for Microsoft's Edge.
+ */
+class Driver extends webdriver.WebDriver {
+  /**
+   * Creates a new browser session for Microsoft's Edge browser.
+   *
+   * @param {(capabilities.Capabilities|Options)=} opt_config The configuration
+   *     options.
+   * @param {remote.DriverService=} opt_service The session to use; will use
+   *     the {@linkplain #getDefaultService default service} by default.
+   * @param {promise.ControlFlow=} opt_flow The control flow to use, or
+   *     {@code null} to use the currently active flow.
+   * @return {!Driver} A new driver instance.
+   */
+  static createSession(opt_config, opt_service, opt_flow) {
+    let executor, service;
+    if (opt_config && opt_config.has('edgeOptions') &&
+      opt_config.get('edgeOptions').getEdgeChromium()) {// chromium edge
```

Comment:
This check should happen after the 'let caps' statement below and should be using the 'caps' variable since we want to check the actual capabilities we are passing to super.createSession.

Also, the key to check is 'ms:edgeChromium', not 'edgeOptions'.

Reply:
added capability key of 'ms:edgeChromium' and added a check in unit test cases.